shared/test/loop_mocha_utils.js

```diff
@@ -83,6 +83,12 @@ var LoopMochaUtils = (function(global, _) {
     });
   };
 
+  syncThenable.resolve = function(result) {
```

Comment:
> Why doesn't this get overwritten by the "this.resolve" dynamic function assignment inside the constructor?

The resolve function doesn't get created until the SyncThenable is created. So, in some places the code is doing `Promise.resolve()` (note: not constructing the promise), this equates to `SyncThenable.resolve()` which doesn't currently exist. Hence, we need to create it.